Add contribex season of docs proposal (#217)
cncf/mentoring: seasonofdocs/README.md

- How to document a new tool
- Bleeding edge command line tools for Kubernetes

### Update the Kubernetes Developer Guide

> The [Kubernetes Developer Guide][kdg] serves as the source of truth for those looking to contribute to the core Kubernetes Project. It along with our [Contributor Guide][kcg] acts the primary means of on-boarding new contributors. While the contributor guide has gone through several large iterations and improvements and is useful for new contributors, our developer documentation has fallen behind and hasn’t been kept in sync with current development practices. This has caused a great deal of friction with potentially new contributors. 

Reviewer: Eliminate the passive voice.
```suggestion
The [Kubernetes Developer Guide][kdg] serves as the source of truth for contributors to the core Kubernetes project. Along with our [Contributor Guide][kcg], the Developer Guide is the primary means of onboarding new contributors. While the Contributor Guide benefits from several large iterations and improvements, and is useful for new contributors, our developer documentation has fallen behind and isn't in sync with current development practices. This causes a great deal of friction with contributors, especially new and prospective ones. 
```

> We are looking for a technical writer who’d like to help with making the developer experience better for all contributors. The developer documentation can be overwhelming, and it’s deficiencies aren’t always obvious to seasoned contributors. We know the developer guide is large, and aren’t expecting a full overhaul -- iterative improvements are a goal.

Reviewer: ```suggestion
We are looking for a technical writer to improve the developer experience for all contributors. The developer documentation can be overwhelming. Its deficiencies aren’t always obvious, even to seasoned contributors. We know the developer guide is large, and aren’t expecting a full overhaul. Our goal is to make iterative improvements.
```

> We can offer a friendly, passionate and supportive [community][kcom] with experienced developers who are available to answer questions and mentor. 

Reviewer: Oxford comma
```suggestion
We are a friendly, passionate, and supportive [community][kcom] with experienced developers who are available to answer questions and mentor a writer.
```

#### Project Goals

> - Revise, improve and update the [Kubernetes Developer Documentation][kdg]

Reviewer: Be consistent with naming
```suggestion
- Revise, improve and update the [Kubernetes Developer Guide][kdg]
```

> - Work closely with our community to establish new development best practices

Reviewer: Everything involves working with the community, no?
```suggestion
- Establish new best practices for developers
```

> - Organize and add metadata to surface the documentation on the [Kubernetes Contributor Site][kcs]

Reviewer: ```suggestion
- Add well-organized metadata to surface the documentation on the [Kubernetes contributor site][kcs]
```

#### What you’ll learn

- How to work with a large diverse and distributed team

> - How content can be managed with the [Hugo site framework][hugo]

Reviewer: ```suggestion
- How to manage content with the [Hugo site framework][hugo]
```

> - Some of the inner workings of the Kubernetes project

Reviewer: ```suggestion
- How to contribute effectively to the Kubernetes project
```

> - Coordinating with active developers who are also working on new APIs and developer features 

Reviewer: ```suggestion
- How to work with open source developers of new APIs and developer features 
```
